### PR TITLE
Misc Egg

### PR DESCRIPTION
**Pull request by elijah-thomas774**

### CODE REVIEW

File: include/common.h [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #define DONT_INLINE __attribute__((never_inline))
 
 // TODO - Fix MSL (stddef.h)
-#define offsetof(ST, M) ((size_t) & (((ST*)0)->M))
+#define offsetof(ST, M) ((size_t) & (((ST *)0)->M))
```
Comment (robojumper) [inline include/common.h:25]: After this is merged we can try to use this macro in the various List users.

File: include/egg/core/eggDecomp.h [hunk 2]
```diff
@@ -2,83 +2,94 @@
 #define EGG_STREAM_DECOMP_H
 
 #include "rvl/CX/cx.h"
+
 #include <common.h>
 
 // NOTE: I put StreamDecomp stuff here. im unsure of the correct place
```
Comment (robojumper) [inline include/egg/core/eggDecomp.h:8]: Maybe adjust this and the header name

File: include/common.h (hunk 1 above)
Comment (elijah-thomas774) [reply include/common.h:25]: Gladly xD

File: include/rvl/CX/cx.h [hunk 3]
```diff
@@ -41,18 +49,18 @@ typedef struct CXUncompContextLRC {
 } CXUncompContextLRC;
 
 // TODO(Zeldex) : Fix Params
```
Comment (robojumper) [inline include/rvl/CX/cx.h:51]:
```suggestion
```

File: src/egg/core/eggDvdRipper.cpp [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+    return pOut;
+}
+
+/* 804949b0 */
+void *DvdRipper::loadToMainRAMDecomp(
+    DvdFile *file, StreamDecomp *decompressor, u8 *pOut, Heap *heap, EAllocDirection allocDir, s32 offset, u32 size,
+    u32 chunkSize, u32 *pRead, u32 *pSize
+) {
+    // TODO Regswaps
+
```
Comment (robojumper) [inline src/egg/core/eggDvdRipper.cpp:115]:
```suggestion
```

**Review by robojumper — APPROVED** [review]
Looks great!

**Pull request merged**